src/store/apiSlice.js

```diff
@@ -6,10 +6,42 @@ export const apiSlice = createApi({
   reducerPath: 'api',
   baseQuery: fetchBaseQuery({ baseUrl: '' }),
   endpoints: (builder) => ({
-    getSources: builder.query({
-      query: () => PROVISIONING_SOURCES_ENDPOINT,
+    getAWSSources: builder.query({
+      async queryFn(_arg, _queryApi, _extraOptions, fetchWithBQ) {
+        // The provisioning sources endpoint response does not include the account ids
+        // associated with the sources. For each source, another API request must be
+        // made to get the account id. This custom queryFn combines all of these API
+        // requests so that the React components can call simply call a single hook,
+        // useGetAWSSourcesQuery().
+
+        const awsSources = await fetchWithBQ(
+          `${PROVISIONING_SOURCES_ENDPOINT}?provider=aws`
+        );
+
+        if (awsSources.error) return { error: awsSources.error };
+
+        const awsAccountIds = await Promise.allSettled(
```

Comment:
I struggled how to decide between using Promise.all() or Promise.allSettled() here... any thoughts?